services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/azure/BlobFileClient.kt

```diff
@@ -84,10 +88,33 @@ class BlobFileClient(blobServiceClient: BlobServiceClient, containers: Container
         return response
     }
 
+    override fun deleteAllByPrefix(container: Container, prefix: String) {
```

Comment:
Suggestion: would it be better if the prefix was of type Number, representing either hanke.id or application.id? 

I guess it would be pretty difficult to find a suitable name for the function deleteAllByX(container: Container, id: Number), but that way the caller would not need to know about the inner functionality of the deletion logic.

Reply:
I think BlobFileClient shouldn't need to know the context where it's being used. The content service decides where the content is saved, so that should be the component that decides what to delete.

Reply:
Fair enough.